lib/iris/tests/unit/coords/test_AuxCoord.py

```diff
@@ -0,0 +1,764 @@
+        self.assertTrue(
+            arrays_share_data(result, self.pts_real),
+            'core_points() do not share data with the internal array.')
+
+    def test_lazy_points(self):
+        lazy_data = self.pts_lazy
+        coord = AuxCoord(lazy_data)
+        result = coord.core_points()
+        self.assertIs(result, lazy_data)
+
+    def test_lazy_points_realise(self):
+        coord = AuxCoord(self.pts_lazy)
+        real_points = coord.points
+        result = coord.core_points()
+        self.assertIs(result, real_points)
+
+
+@skip_lazy_auxcoords
+class Test_core_bounds(tests.IrisTest):
+    # Test for AuxCoord.core_bounds() with various types of points and bounds.
+    def setUp(self):
+        setup_test_arrays(self)
+
+    def test_no_bounds(self):
+        coord = AuxCoord(self.pts_real)
+        result = coord.core_bounds()
+        self.assertIsNone(result)
+
+    def test_real_bounds(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_real)
+        result = coord.core_bounds()
+        self.assertArrayEqual(result, self.bds_real)
+        self.assertTrue(
+            arrays_share_data(result, self.bds_real),
+            'core_bounds() do not share data with the internal array.')
+
+    def test_lazy_bounds(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_lazy)
+        result = coord.core_bounds()
+        self.assertIsInstance(result, da.Array)
+        self.assertEqual(result.dtype, self.bds_real.dtype)
+        self.assertArrayEqual(result.compute(), self.bds_real)
+
+    def test_lazy_bounds_realise(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_lazy)
+        real_bounds = coord.bounds
+        result = coord.core_bounds()
+        self.assertIs(result, real_bounds)
+
+
+@skip_lazy_auxcoords
+class Test_lazy_points(tests.IrisTest):
+    def setUp(self):
+        setup_test_arrays(self)
+
+    def test_real_core(self):
+        coord = AuxCoord(self.pts_real)
+        result = coord.lazy_points()
+        self.assertIsInstance(result, da.Array)
+        self.assertArrayEqual(result.dtype, self.pts_real.dtype)
+        self.assertArrayEqual(result.compute(), self.pts_real)
+
+    def test_lazy_core(self):
+        coord = AuxCoord(self.pts_lazy)
+        result = coord.lazy_points()
+        self.assertIs(result, self.pts_lazy)
+
+
+@skip_lazy_auxcoords
+class Test_lazy_bounds(tests.IrisTest):
+    def setUp(self):
+        setup_test_arrays(self)
+
+    def test_real_core(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_real)
+        result = coord.lazy_bounds()
+        self.assertIsInstance(result, da.Array)
+        self.assertArrayEqual(result.dtype, self.bds_real.dtype)
+        self.assertArrayEqual(result.compute(), self.bds_real)
+
+    def test_lazy_core(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_lazy)
+        result = coord.lazy_bounds()
+        self.assertIs(result, self.bds_lazy)
```

Comment:
Time for a controversial question: should you include a test for no bounds here too? Evidently, if the coord has no bounds then `lazy_bounds` will be `None`. I think there should be a test for that here, but if we do that then we may need to add equivalent tests elsewhere in these test modules...